test/Conversion/TorchToLinalg/pooling.mlir

```diff
@@ -90,8 +90,13 @@ func.func @forward_max_pool3d(%arg0: !torch.vtensor<[?,?,?,?,?],f32>) -> !torch.
   // CHECK: %[[OUTPUT_TENSOR:.*]] = linalg.fill ins(%[[MIN_VALUE:.*]] : f32) outs(%{{.*}} : tensor<?x?x?x?x?xf32>) -> tensor<?x?x?x?x?xf32>
   // CHECK: %[[MAX_3D_POOL:.*]] = linalg.generic {indexing_maps = [#map, #map1, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel", "parallel", "reduction", "reduction", "reduction"]} ins(%[[PADDED_INPUT_TENSOR:.*]], %{{.*}} : tensor<?x?x?x?x?xf32>, tensor<8x8x8xf32>) outs(%[[OUTPUT_TENSOR:.*]] : tensor<?x?x?x?x?xf32>) {
   // CHECK-NEXT:  ^bb0(%[[CURRENT_VALUE:.*]]: f32, %[[KERNEL:.*]]: f32, %[[ACC_OUT:.*]]: f32):
-  // CHECK-NEXT:    %[[MAXF:.*]] = arith.maximumf %[[CURRENT_VALUE:.*]], %[[ACC_OUT:.*]] : f32
-  // CHECK-NEXT:    linalg.yield %[[MAXF:.*]] : f32
+  // CHECK:         %[[MAX_PRED:.*]] = arith.cmpf ugt, %[[CURRENT_VALUE]], %[[ACC_OUT]] : f32
+  // CHECK:         %[[MAXF:.*]] = scf.if %[[MAX_PRED]] -> (f32) {
+  // CHECK:           scf.yield %[[CURRENT_VALUE]] : f32
+  // CHECK:         } else {
+  // CHECK:           scf.yield %[[ACC_OUT]] : f32
+  // CHECK:         }
+  // CHECK-NEXT:    linalg.yield %[[MAXF]] : f32
```

Comment:
@lingzhiz1998 I haven't gone through the code in this PR, but this IR getting generated instead of the earlier one, is not good for the downstream pipeline, and I don't see a point in doing that.

Reply:
This PR aims to change the IR of max_pool _with_indices op to eliminate extract op. Generating different IRs for max_pool op is not necessary. The reason of why I did this is making the lowering logic of these ops as identical as possible. If you worry about this, I can keep the IR of max_pool op as same as previous.